libs/ui-watt/src/lib/components/text-field/watt-text-field.stories.ts

```diff
@@ -0,0 +1,164 @@
+
+export const Input: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'Input',
+    exampleFormControl: new FormControl(null),
+  },
+  template,
+});
+
+Input.parameters = {
+  docs: {
+    source: {
+      code: howToUseGuideBasic,
+    },
+  },
+};
+
+export const WithDisabled: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'disabled',
+    exampleFormControl: new FormControl({ value: null, disabled: true }),
+  },
+  template,
+});
+
+WithDisabled.parameters = {
+  docs: {
+    source: {
+      code: `new FormControl({ value: null, disabled: true })`,
+    },
+  },
+};
+
+export const WithValue: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'withValue',
+    exampleFormControl: new FormControl({ value: 'this is a value', disabled: false }),
+  },
+  template,
+});
+
+WithValue.parameters = {
+  docs: {
+    source: {
+      code: `new FormControl({ value: 'this is a value', disabled: false })`,
+    },
+  },
+};
+
+export const WithPlaceholder: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'withPlaceholder',
+    exampleFormControl: new FormControl(null),
+    placeholder: "I'm a placeholder",
+  },
+  template,
+});
+
+export const WithNumber: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'number',
+    exampleFormControl: new FormControl(null),
+    type: 'number',
+  },
+  template,
+});
+
+export const WithRequired: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'required',
+    exampleFormControl: new FormControl(null),
+  },
+  template: `<watt-text-field [required]="true" [label]="label" [type]="type" [placeholder]="placeholder" [formControl]="exampleFormControl">
+              <span info *ngIf="exampleFormControl.dirty && exampleFormControl.hasError('required')">This field is required</span>
+            </watt-text-field>`,
+});
+
+export const WithPrefix: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'Prefix',
+    exampleFormControl: new FormControl(null),
+  },
+  template: `<watt-text-field prefix="search" [label]="label" [required]="required" [type]="type" [placeholder]="placeholder" [formControl]="exampleFormControl" />`,
+});
+
+export const WithSuffix: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'Suffix',
+    exampleFormControl: new FormControl(null),
+  },
+  template: `<watt-text-field [label]="label" [required]="required" [type]="type" [placeholder]="placeholder" [formControl]="exampleFormControl">
+              <watt-button variant="icon" icon="search" />
+             </watt-text-field>`,
+});
+
+export const WithHint: StoryFn<WattTextFieldComponent> = () => ({
+  props: {
+    label: 'hint',
+    exampleFormControl: new FormControl(null),
+  },
+  template: `<watt-text-field [required]="true" [label]="label" [type]="type" [placeholder]="placeholder" [formControl]="exampleFormControl">
+              <span error>This field is required</span>
```

Comment:
question: do we want to support two ways of showing the error message - (1) by using an `ngIf` on the error (like the `WithRequired` story) (which adds/removes the error element from the DOM) and (2) by simply adding the `error` attribute on an element (so the element is shown/hidden with CSS)?

Reply:
Only with an error attribute, i forgot to remove it :-)